scripts/west_commands/blobs.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import hashlib
+import importlib
+import os
+from pathlib import Path
+import re
+import sys
+import textwrap
+
+from abc import ABC, abstractmethod
+from urllib.parse import urlparse
+from west import log
+from west.commands import WestCommand
+
+from fetchers.fetcher import fetchersmeta
+from zephyr_ext_common import ZEPHYR_BASE
+
+sys.path.append(os.fspath(Path(__file__).parent.parent))
+import zephyr_module
+
+class Blobs(WestCommand):
+
+    def __init__(self):
+        super().__init__(
+            'blobs',
+            # Keep this in sync with the string in west-commands.yml.
+            'work with binary blobs',
+            'Work with binary blobs',
+            accepts_unknown_args=False)
+
+    def do_add_parser(self, parser_adder):
+        default_fmt = '{module}: {status} {path} ({type}, {abspath})'
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description,
+            epilog=textwrap.dedent(f'''\
+            FORMAT STRINGS
+            --------------
+
+            Blobs are listed using a Python 3 format string. Arguments
+            to the format string are accessed by name.
+
+            The default format string is:
+
+            "{default_fmt}"
+
+            The following arguments are available:
+
+            - module: name of the module that contains this blob
+            - abspath: blob absolute path
+            - status: short status (A: present, M: hash failure, D: not present)
+            - path: blob local path from <module>/zephyr/blobs/
+            - sha256: blob SHA256 hash in hex
+            - type: type of blob
+            - version: version string
+            - license_path: path to the license file for the blob
+            - uri: URI to the remote location of the blob
+            - description: blob text description
+            - doc-url: URL to the documentation for this blob
+            '''))
+
+        # Remember to update west-completion.bash if you add or remove
+        # flags
+        parser.add_argument('subcmd', nargs=1, choices=['list', 'fetch'],
+                            help='''Select the sub-command to execute.
+                            Currently only list and fetch are supported.''')
+
+        # Remember to update west-completion.bash if you add or remove
```

Comment:
Do you plan on doing this?

Reply:
Yes, but I will do a wholesale update of the completion script in a follow-up PR.